eogrow/utils/validators.py

```diff
@@ -91,6 +91,18 @@ def ensure_both(cls: type, value: Optional[Any], values: RawSchemaDict) -> Optio
     return field_validator(second_param, ensure_both, **kwargs)
 
 
+def ensure_storage_key_presence(key: str, **kwargs: Any) -> classmethod:
+    """A field validator that makes sure that the specified storage key is present in the storage structure."""
+
+    def validate_storage_key(cls: type, key: Optional[str], values: RawSchemaDict) -> Optional[str]:
```

Comment:
a possible ouchie (that likely cant happen, but i want a test to show that it doesnt) is that you have a  `input_folder: str` but then this validator let's a `None` through...
Very likely pydantic doublechecks the final type, but I want guarantees in form of tests

Reply:
This would be an issue if we would use the validator with `pre=True`, right? Here the parameter value `None` should fail even before the validation is reached.

having a test for this feels like testing the pydantic model itself, not the validator.

This is just my thinking, if you think it's still better to have an extra test for this, I'll gladly add it.

Otherwise I also tried this and I get what one would expect:

```
E   pydantic.error_wrappers.ValidationError: 1 validation error for DummySchema
E   folder_key
E     none is not an allowed value (type=type_error.none.not_allowed)
```

Reply:
added